clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -323,11 +323,17 @@ private synchronized void disableHeartbeatThread() {
     }
 
     private void closeHeartbeatThread() {
```

Comment:
It should always be called from the same thread, though we've taken a simple approach to locking in this class where everything is protected as far as I can tell.

This does technically introduce a behavioral change though, since `heartbeatThread` is being reset to `null` whereas it was not before, but this doesn't seem like a problem.